### ISSUE TITLE
[FEAT] Extend auction contracts to support staking functionality

### ISSUE THREAD

**Issue by devendran-m** [casper-network/casper-node#4929]
Feature Addition:

- Extend the auction contract to track stakes placed by smart contracts.
- Allow smart contracts to submit stake amounts and specify parameters such as validators to stake with.
- Support contract-based delegation of stakes to validators, where multiple smart contracts can delegate stakes to the same validator.

**Comment by EdHastingsCasperAssociation** [comment]
I've worked on this for about 7 days so far...initially things were promising, but the code base has just changed too much since the original thought was put into this feature. I eventually ran into problems with the new delegator reservations feature which I overcame, but then ran into an issue migrating unbonds. More or less got that worked out, but messily. Then ran into issues with redelgation, auctionstate reporting, and some retrocompat issues. 

Long story short, its just getting out of hand. I think this feature requires more thought and updated design work that takes into consideration the current state of the code base. I'd like to consider punting this to a later release.
